e2e/pkg/tests/networking/maglev.go

```diff
@@ -0,0 +1,588 @@
+
+			// Assert that the backend selected via node 3 is the same as that selected via node 2
+			Expect(backendViaNode3IPv4).Should(Equal(backendViaNode2IPv4),
+				fmt.Sprintf("Expected IPv4 backend selection to be consistent across nodes: node 2 selected %s, node 3 selected %s",
+					backendViaNode2IPv4, backendViaNode3IPv4))
+
+			Expect(backendViaNode3IPv6).Should(Equal(backendViaNode2IPv6),
+				fmt.Sprintf("Expected IPv6 backend selection to be consistent across nodes: node 2 selected %s, node 3 selected %s",
+					backendViaNode2IPv6, backendViaNode3IPv6))
+
+			framework.Logf("Maglev cross-node consistency verified: IPv4 backend %s and IPv6 backend %s selected consistently via nodes 2 and 3",
+				backendViaNode2IPv4, backendViaNode2IPv6)
+		})
+	})
+
+type MaglevTests struct {
+	f                   *framework.Framework
+	serviceClusterIPv4  string
+	serviceClusterIPv6  string
+	loadBalancerService *v1.Service
+	maglevConfig        *MaglevConfig
+	nodeNameToIPv4      map[string]string
+	nodeNameToIPv6      map[string]string
+	connTester          conncheck.ConnectionTester
+}
+
+type MaglevConfig struct {
+	ServiceName      string
+	ServicePort      int32
+	SourcePort       int // Source port for requests to test consistent hashing
+	NumberOfRequests int // Number of requests to send to the cluster IP for testing
+	IntervalSeconds  int // Interval between requests in seconds
+}
+
+func NewMaglevTests(f *framework.Framework) *MaglevTests {
+	return &MaglevTests{
+		f: f,
+		maglevConfig: &MaglevConfig{
+			ServiceName:      "netexec",
+			ServicePort:      8080,
+			SourcePort:       12345, // Fixed source port for consistent hashing tests
+			NumberOfRequests: 10,    // Default number of requests for testing
+			IntervalSeconds:  1,     // Default interval between requests
+		},
+		nodeNameToIPv4: make(map[string]string),
+		nodeNameToIPv6: make(map[string]string),
+	}
+}
+
+// parseBackendResponse parses the JSON response from netexec and returns the backend pod name
+func (m *MaglevTests) parseBackendResponse(output string) (string, error) {
+	// NetexecResponse represents the JSON response from the netexec service
+	type NetexecResponse struct {
+		Output string `json:"output"`
+	}
+
+	var response NetexecResponse
+	if err := json.Unmarshal([]byte(output), &response); err != nil {
+		return "", fmt.Errorf("failed to parse JSON response: %w", err)
+	}
+
+	// Remove the trailing newline from the output
+	backendName := strings.TrimSpace(response.Output)
+
+	// Verify this matches our expected backend pod naming pattern (backend-pod-0 to backend-pod-19)
+	backendPodPattern := regexp.MustCompile(`^backend-pod-\d+$`)
+	if !backendPodPattern.MatchString(backendName) {
+		return "", fmt.Errorf("response '%s' does not match expected backend pod naming pattern 'backend-pod-N'", backendName)
+	}
+
+	return backendName, nil
+}
+
+// SetSourcePort updates the source port used for curl requests
+func (m *MaglevTests) SetSourcePort(port int) {
```

Comment:
Are these setter functions called from anywhere? I don't see references to them in this PR as far as I can tell.